src/Views/dialogView.cpp

```diff
@@ -0,0 +1,242 @@
+#include "dialogView.h"
+
+#define DIALOG_WIDTH 760
+#define DIALOG_HEIGHT 440
+#define DIALOG_PADDING_X 79
+#define DIALOG_PADDING_Y 30
+#define DIALOG_BTN_WIDTH 322
+#define DIALOG_BTN_HEIGHT 56
+#define DIALOG_BTN_PADDING 5
+#define DIALOG_MSG_HEIGHT 315
+#define DIALOG_X ((SCREEN_WIDTH-DIALOG_WIDTH)/2)
+#define DIALOG_Y ((SCREEN_HEIGHT-DIALOG_HEIGHT)/2)
+#define DIALOG_BTN_Y (DIALOG_Y+365)
+#define DIALOG_BTN_INNER_TOP_Y (DIALOG_BTN_Y+DIALOG_BTN_PADDING)
+#define DIALOG_BTN_INNER_BOT_Y (DIALOG_BTN_Y+DIALOG_BTN_HEIGHT-DIALOG_BTN_PADDING)
+#define DIALOG_BTN_1_OF_1_X (DIALOG_X+DIALOG_WIDTH/2-DIALOG_BTN_WIDTH/2)
+#define DIALOG_BTN_1_OF_1_INNER_LEFT_X (DIALOG_BTN_1_OF_1_X+DIALOG_BTN_PADDING)
+#define DIALOG_BTN_1_OF_1_INNER_RIGHT_X (DIALOG_BTN_1_OF_1_X+DIALOG_BTN_WIDTH-DIALOG_BTN_PADDING)
+#define DIALOG_BTN_1_OF_2_X (DIALOG_X+44)
+#define DIALOG_BTN_1_OF_2_INNER_LEFT_X (DIALOG_BTN_1_OF_2_X+DIALOG_BTN_PADDING)
+#define DIALOG_BTN_1_OF_2_INNER_RIGHT_X (DIALOG_BTN_1_OF_2_X+DIALOG_BTN_WIDTH-DIALOG_BTN_PADDING)
+#define DIALOG_BTN_2_OF_2_X (DIALOG_X+394)
+#define DIALOG_BTN_2_OF_2_INNER_LEFT_X (DIALOG_BTN_2_OF_2_X+DIALOG_BTN_PADDING)
+#define DIALOG_BTN_2_OF_2_INNER_RIGHT_X (DIALOG_BTN_2_OF_2_X+DIALOG_BTN_WIDTH-DIALOG_BTN_PADDING)
+#define DIALOG_FOCUS_GLOW_HALF_CYCLE_FRAMES 45
+
+
+extern unsigned char _binary_assets_spr_img_dialog_msg_bg_png_start;
+extern unsigned char _binary_assets_spr_img_dialog_msg_btn_png_start;
+extern unsigned char _binary_assets_spr_img_dialog_msg_btn_active_png_start;
+extern unsigned char _binary_assets_spr_img_dialog_msg_btn_focus_png_start;
+
+DialogView::DialogView() :
+	msg_font(Font(std::string(FONT_DIR "segoeui.ttf"), 28)),
+	btn_font(Font(std::string(FONT_DIR "segoeui.ttf"), 25)),
+	img_dialog_msg_bg(Texture(&_binary_assets_spr_img_dialog_msg_bg_png_start)),
+	img_dialog_msg_btn(Texture(&_binary_assets_spr_img_dialog_msg_btn_png_start)),
+	img_dialog_msg_btn_active(Texture(&_binary_assets_spr_img_dialog_msg_btn_active_png_start)),
+	img_dialog_msg_btn_focus(Texture(&_binary_assets_spr_img_dialog_msg_btn_focus_png_start))
+{
+	priority = 550;
+	log_printf(DBG_DEBUG, "DialogView::DialogView()");
+	// FIXME HACK: when DialogView is passed to Activity::AddView() it's destroyed once the activity is crashed
+	// Keeping an internal shared_ptr of itself makes sure that it's never destroyed
+	me_ptr = std::shared_ptr<DialogView>(this);
+}
+
+DialogView::~DialogView() {
+	log_printf(DBG_WARNING, "DialogView destructor called");
+}
+
+void DialogView::openDialogView(DialogViewResult *result, std::string message, DialogType type) {
```

Comment:
Please use `std::shared_ptr<DialogViewResult>` instead just like `IMEView` does for `IMEViewResult`.